drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -121,18 +121,19 @@ static int lis2dh_sample_fetch(struct device *dev, enum sensor_channel chan)
 {
 	struct lis2dh_data *lis2dh = dev->driver_data;
 	size_t i;
-	int status;
+	int status = 0;
 
 	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL ||
 			chan == SENSOR_CHAN_ACCEL_XYZ);
 
 	/*
-	 * since status and all accel data register addresses are consecutive,
-	 * a burst read can be used to read all the samples
+	 * Burst mode cannot be used with this sensor.
 	 */
-	status = lis2dh_burst_read(dev, LIS2DH_REG_STATUS,
-				   lis2dh->sample.raw,
-				   sizeof(lis2dh->sample.raw));
+	for(i = 0; i < sizeof(lis2dh->sample.raw); i++) {
+		status |= lis2dh_reg_read_byte(lis2dh->bus,
+			LIS2DH_REG_STATUS + i, &lis2dh->sample.raw[i]);
+	}
+
 	if (status < 0) {
 		SYS_LOG_WRN("Could not read accel axis data");
 		return status;
```

Comment:
Actually I think that the sensor works fine with I2C burst transaction.
What you have to do is simply set the I2C AUTOINCREMENT bit, like the lis3dh driver does. If you see here in this driver the bit is defined (LIS2DH_AUTOINCREMENT_ADDR) but is not used. Take a look to lis3dh driver and of course to the lis2dh datasheet.

Reply:
@ekalugin 
You have to do something like:
status = lis2dh_burst_read(dev, LIS2DH_REG_STATUS | LIS2DH_AUTOINCREMENT_ADDR, .....

Reply:
Pls let me know if it works

Reply:
I report the LIS2DH datasheet section that explains the I2C auto-increment address:

The I2C embedded inside the LIS2DH behaves like a slave device and the following protocol must be adhered to. After the start condition (ST) a slave address is sent, once a slave acknowledge (SAK) has been returned, a 8-bit sub-address (SUB) is transmitted: the 7 LSb represent the actual register address while the MSB enables address auto increment. If the MSb of the SUB field is ‘1’, the SUB (register address) is automatically increased to allow multiple data read/write.